roxctl/README.md

```diff
@@ -0,0 +1,189 @@
+type sampleCmd struct {
+	env environment.Environment
+
+	name string
+}
+
+// Construct ensure all flag values are resolved and possibly validated. Any additional external dependency that you may
+// require (e.g. a gRPC client) may be constructed here as well.
+func (s *sampleCmd) Construct(cmd *cobra.Command) error {
+	return nil
+}
+
+// Sample is an example for any possible business logic your command may execute.
+func (s *sampleCmd) sample() error {
+	s.env.Logger().InfofLn("Hello world, %s!", s.name)
+	return nil
+}
+```
+
+# Maintaining commands
+
+One of the most important things about maintaining commands for `roxctl` is to ensure changes of behavior have no side
+effects.
+
+Since `roxctl` is typically used within CI as well as automation (e.g. scripts) by users, it is sensitive to behavior
+changes.
+
+A behavior change may consist of:
+- Changing the default value of a flag.
+- Removing or renaming a flag.
+- Removing or renaming a command.
+- Changing non-human readable CLI output (e.g. changing returned JSON).
+
+Since this is a sensitive topic, the following points **must** be considered when maintaining commands:
+- Breaking changes **should** be avoided if possible.
+- A breaking change **must** have an intrinsic value to it. While this may be vague, here are a couple of examples:
+  - Allows to introduce new functionality which provides direct value to a majority of users.
+  - Increases stability or usability for a majority of users.
+  - Reduces the overall maintenance cost (e.g. removing unused commands, removing support for deprecated
+    OpenShift versions)
+- When breaking changes are done, a deprecation notice within the `CHANGELOG.md` **must** exist beforehand
+  according to our current guidelines for introducing breaking changes.
+- When removing or renaming a command or flag, an alias **must** be introduced (see the latter section about this).
+
+## Creating an alias for deprecated values
+
+When changing flags or commands, aliases must be in place to avoid breaking users of said command.
+
+Currently, two options exist for this:
+- Cobra's [alias feature](https://pkg.go.dev/github.com/spf13/cobra#Command).
+- Handling the alias in-code.
+
+For Cobra's alias feature, the following limitations currently apply:
+- Renaming a command is straight-forward **as long as the command hierarchy does not change**.
+- Creating an alias for a flag currently is not supported, only for args.
+
+In case your change falls under the limitations of the cobra alias feature, it's recommended to handle the alias within
+the command's code itself.
+
+The below code highlights changes for a command:
+```go
+package sample
+
+func OldCommand(cliEnvironment environment.Environment) *cobra.Command {
+	sampleCmd := &sampleCmd{env: cliEnvironment}
+
+	cmd := &cobra.Command{
+		Use: "old-sample",
+		Short: "A sample command",
+		Long: `A sample command.
```

Comment:
Shall descriptions be updated to mention that the old command is deprecated and what to use instead?

Reply:
Added this to the `Long` description, with `Hidden` set to true, you'll only ever see the command's help if you do `roxctl old-sample --help`.